### ISSUE TITLE
nox session coverage-3.6 fails with python <= 3.6.3

### ISSUE THREAD

**Issue by odelalleau** [omry/omegaconf#314]
Repro steps:

```
conda create -n py363-tmp python=3.6.3 --yes
conda activate py363-tmp
pytest tests/test_base_config.py::TestCopy
```

Observe multiple errors like:
```
FAILED tests/test_base_config.py::TestCopy::test_copy[src0-<lambda>0] - TypeError: __copy__() takes 1 positional argument but 2 were given
```

The error is related to this item in Python 3.6.4 changelog (https://docs.python.org/3.6/whatsnew/changelog.html#python-3-6-4-final):
```
bpo-28556: Two minor fixes for typing module: allow shallow copying instances of generic classes, [...]
```

I ran into this because on my MacBook `nox` is re-using `/opt/homebrew/opt/python36/bin/python3.6` which is 3.6.2.

I couldn't find a way to force `nox` to use Python >= 3.6.4 so I hacked around a PR to just skip these tests when this happens (will submit it shortly).

**Comment by omry** [comment]
I hereby declare 3.6.3 unsupported.
See my comment on the PR for how to skip this test for it.

**Comment by omry** [comment]
On a side note:
1. I recommend that you use Conda and not the system interpreter.
2. I recommend that you use Python 3.8 unless you are debugging issues specific to older versions. the CI will protect you from accidentally relying on too-new features.

**Comment by omry** [comment]
> I couldn't find a way to force nox to use Python >= 3.6.4 so I hacked around a PR to just skip these tests when this happens (will submit it shortly).

To clarify:
install nox inside your chosen conda environment. it should use the interpreter from it.
if it does not, you may need to restart your shell.

**Comment by odelalleau** [comment]
> On a side note:
> 
> 1. I recommend that you use Conda and not the system interpreter.
> 2. I recommend that you use Python 3.8 unless you are debugging issues specific to older versions. the CI will protect you from accidentally relying on too-new features.

Yeah I'm actually using Conda with Python 3.8. It's just that when I simply run `nox` to run the full test suite to check that everything is ok, for some reason nox seems to be keen to re-use my system Python 3.6 for the 3.6 tests.

Thanks for merging the PR! Will make it easier for me to test locally :)

**Comment by omry** [comment]
About 3.6:
If the only available Python 3.6 nox can find in the system one it will use it.
see if you can upgrade it.
Generally I create a conda environment for each Python version:
```
$ conda env list | grep omegaconf
omegaconf-pypy36         /home/omry/miniconda3/envs/omegaconf-pypy36
omegaconf36              /home/omry/miniconda3/envs/omegaconf36
omegaconf37              /home/omry/miniconda3/envs/omegaconf37
omegaconf38              /home/omry/miniconda3/envs/omegaconf38
```
When testing, you can tell nox to run just a specific session nox -s (see nox -l for session lists).
